src/stackAnalysisService.ts

```diff
@@ -3,18 +3,27 @@
 import * as vscode from 'vscode';
 import * as request from 'request';
 
+const MAX_RETRIES = 5;
+const INITIAL_DELAY = 1000; // in milliseconds
+
 export module stackAnalysisServices {
   export const clearContextInfo = context => {
     context.globalState.update('f8_3scale_user_key', '');
     context.globalState.update('f8_access_routes', '');
   };
 
-  export const getStackAnalysisService = options => {
+  export const getStackAnalysisService = (options, retryCount = 0) => {
     let errorMsg: string;
-    return new Promise((resolve, reject) => {
+
+    const getRequestWithExponentialBackoff = (resolve, reject) => {
       request.get(options, (err, httpResponse, body) => {
         if (err) {
-          reject(err);
+          if (retryCount < MAX_RETRIES) {
+            retryCount++;
+            invokeExponentialBackoff(retryCount, getRequestWithExponentialBackoff, resolve, reject);
+          } else {
+            reject(err);
+          }
         } else {
           if (
             httpResponse.statusCode === 200 ||
```

Comment:
is there any chance backend will send back 202 or 201?

Reply:
for status code 202, that is already included in the if statement 
```
if (
            httpResponse.statusCode === 200 ||
            httpResponse.statusCode === 202
    )
```

as for status code 201, HTTP status code 201 is specifically used to indicate that a POST, PUT, or PATCH request has been successful and that a new resource has been created. A GET request would not likely return a status code of 201. 
according to this, the method getStackAnalysisService does not require status 201 to be considered.
On the other hand, method postStackAnalysisService could take status code 201 into consideration.

But judging by the fact that the creators of this code did not include status code 201 to begin with, can we be confident that they considered their API to never return such a status code?

Reply:
Agreed.